src/analysis/complex/exponential.lean

```diff
@@ -1885,3 +1900,111 @@ lemma has_deriv_within_at.rexp {f : ℝ → ℝ} {f' x : ℝ} {s : set ℝ}
   (hf : has_deriv_within_at f f' s x) :
   has_deriv_within_at (real.exp ∘ f) (f' * real.exp (f x)) s x :=
 (real.has_deriv_at_exp (f x)).comp_has_deriv_within_at x hf
+
+namespace nnreal
+
+/-- The nonnegative real power function `x^y`, defined for `x : nnreal` and `y : ℝ ` as the
+restriction of the real power function. For `x > 0`, it is equal to `exp (y log x)`. For `x = 0`,
+one sets `0 ^ 0 = 1` and `0 ^ y = 0` for `y ≠ 0`. -/
+noncomputable def rpow (x : nnreal) (y : ℝ) : nnreal :=
+⟨(x : ℝ) ^ y, real.rpow_nonneg_of_nonneg x.2 y⟩
+
+noncomputable instance : has_pow nnreal ℝ := ⟨rpow⟩
```

Comment:
Does it make sense to have `nnreal.rpow`, not just `nnreal.has_pow`? I mean, all lemmas deal with `has_pow`, so if someone will try to use `nnreal.rpow` function in a lemma, (s)he might be surprised by `rw`/`simp` failures.

Reply:
I followed the design of the other power functions in this file. But I agree with you that it might be calling for trouble. To mitigate this, I have added simp lemmas turning `rpow x y` into `x ^ y`, here and at other places in this file.